File: packages/go/courses.go
package hubfly

import (
	"context"
	"fmt"
)

type CoursesService struct {
	client *Client
}

func (s *CoursesService) GetAssignments(ctx context.Context, courseID string) (*Response[[]interface{}], error) {
	path := fmt.Sprintf("/api/v1/courses/%s/assignments", courseID)
	return Request[[]interface{}](ctx, s.client, "GET", path, nil)
}

func (s *CoursesService) CreateAssignment(ctx context.Context, courseID string, body interface{}) (*Response[interface{}], error) {
	path := fmt.Sprintf("/api/v1/courses/%s/assignments/create", courseID)
	return Request[interface{}](ctx, s.client, "POST", path, body)
}

File: packages/ts/src/modules/courses.ts
import type { HubflyClient } from '../client.js';
import type { ApiResponse } from '../types.js';

export class CoursesModule {
  constructor(private readonly client: HubflyClient) {}

  public async getAssignments(courseId: string): Promise<ApiResponse<unknown[]>> {
    return this.client.request<unknown[]>(`/api/v1/courses/${courseId}/assignments`);
  }

  public async createAssignment(courseId: string, params: unknown): Promise<ApiResponse<unknown>> {
    return this.client.request<unknown>(`/api/v1/courses/${courseId}/assignments/create`, {
      method: 'POST',
      body: params,
    });
  }
}
